apps/agent/test/e2e/scenarios/01_happy_path/index.spec.ts

```diff
@@ -625,4 +635,357 @@ describe.sequential("single agent", () => {
+
+        // A1 creates a request REQ1 for Epoch1
+        await protocolProvider.createRequest(currentEpoch.number, PROTOCOL_L2_CHAIN_ID);
+
+        const requestCreatedEvent = await waitForEvent({
+            client: anvilClient,
+            filter: {
+                address: protocolContracts["Oracle"],
+                fromBlock: initBlock,
+                event: getAbiItem({ abi: oracleAbi, name: "RequestCreated" }),
+                strict: true,
+            },
+            pollingIntervalMs: 100,
+            blockTimeout: initBlock + 1000n,
+        });
+
+        expect(requestCreatedEvent).toBeDefined();
+
+        const correctResponse = await blockNumberService.getEpochBlockNumber(
+            currentEpoch.startTimestamp,
+            PROTOCOL_L2_CHAIN_ID,
+        );
+
+        // A1 proposes a response RESP1(REQ1)
+        await protocolProvider.proposeResponse(requestCreatedEvent.args._request, {
+            proposer: accounts[0].account.address,
+            requestId: requestCreatedEvent.args._requestId as RequestId,
+            response: ProphetCodec.encodeResponse({ block: correctResponse }),
+        });
+
+        const responseProposedEvent = await waitForEvent({
+            client: anvilClient,
+            filter: {
+                address: protocolContracts["Oracle"],
+                fromBlock: initBlock,
+                event: getAbiItem({ abi: oracleAbi, name: "ResponseProposed" }),
+                strict: true,
+            },
+            pollingIntervalMs: 100,
+            blockTimeout: initBlock + 1000n,
+        });
+
+        expect(responseProposedEvent).toBeDefined();
+
+        // Setting up a second account to act as the disputer and pledger
+        const account2 = accounts[1];
+
+        // Account2 disputes RESP1, creating DISP1
+        const protocolProviderAccount2 = new ProtocolProvider(
+            {
+                l1: {
+                    chainId: PROTOCOL_L2_CHAIN_ID,
+                    urls: [PROTOCOL_L2_LOCAL_URL],
+                    transactionReceiptConfirmations: 1,
+                    timeout: 1_000,
+                    retryInterval: 500,
+                },
+                l2: {
+                    chainId: PROTOCOL_L2_CHAIN_ID,
+                    urls: [PROTOCOL_L2_LOCAL_URL],
+                    transactionReceiptConfirmations: 1,
+                    timeout: 1_000,
+                    retryInterval: 500,
+                },
+            },
+            {
+                bondEscalationModule: protocolContracts["BondEscalationModule"],
+                eboRequestCreator: protocolContracts["EBORequestCreator"],
+                epochManager: EPOCH_MANAGER_ADDRESS,
+                oracle: protocolContracts["Oracle"],
+                horizonAccountingExtension: protocolContracts["HorizonAccountingExtension"],
+            },
+            account2.privateKey,
+            blockNumberService,
+        );
+
+        const dispute = {
```

Comment:
This is just a side-note for future tests, but I found that it was super useful to explicitly declare the type of the variables to get some ts help when building the object:
```suggestion
        const dispute: Dispute["prophetData"] = {
```